ophyd/signal.py

```diff
@@ -392,6 +305,201 @@ def get(self, as_string=None, **kwargs):
+
+        Returns
+        -------
+        dict
+            Dictionary of name and formatted description string
+        """
+        return {self.name: {'source': 'PV:{}'.format(self._read_pv.pvname),
+                            'dtype': 'number',
+                            'shape': []}}
+
+    @raise_if_disconnected
+    def read(self):
+        """Read the signal and format for data collection
+
+        Returns
+        -------
+        dict
+            Dictionary of value timestamp pairs
+        """
+
+        return {self.name: {'value': self.value,
+                            'timestamp': self.timestamp}}
+
+
+class EpicsSignalRO(EpicsSignalBase):
+    '''A read-only EpicsSignal -- that is, one with no `write_pv`
+
+    Keyword arguments are passed on to the base class (Signal) initializer
+
+    Parameters
+    ----------
+    read_pv : str
+        The PV to read from
+    pv_kw : dict, optional
+        Keyword arguments for epics.PV(**pv_kw)
+    limits : bool, optional
+        Check limits prior to writing value
+    auto_monitor : bool, optional
+        Use automonitor with epics.PV
+    name : str, optional
+        Name of signal.  If not given defaults to read_pv
+    '''
+    def __init__(self, read_pv, *, pv_kw=None, string=False, auto_monitor=None,
+                 name=None, **kwargs):
+        super().__init__(read_pv, pv_kw=pv_kw, string=string,
+                         auto_monitor=auto_monitor, name=name,
+                         separate_readback=False, **kwargs)
+
+    # TODO: need to further refactor Signal to SignalBase, Signal, SignalRO, to
+    #       remove these methods
+    def put(self, *args, **kwargs):
+        raise ReadOnlyError('Read-only signals cannot be put to')
+
+    def get_setpoint(self, *args, **kwargs):
+        raise ReadOnlyError('EpicsSignalRO has no setpoint')
+
+    @property
+    def setpoint_ts(self, *args, **kwargs):
+        raise ReadOnlyError('EpicsSignalRO has no setpoint')
+
+
+class EpicsSignal(EpicsSignalBase):
+    '''An EPICS signal, comprised of either one or two EPICS PVs
+
+    Keyword arguments are passed on to the base class (Signal) initializer
+
+    Parameters
+    ----------
+    read_pv : str
+        The PV to read from
+    write_pv : str, optional
+        The PV to write to if different from the read PV
+    pv_kw : dict, optional
+        Keyword arguments for epics.PV(**pv_kw)
+    limits : bool, optional
+        Check limits prior to writing value
+    auto_monitor : bool, optional
+        Use automonitor with epics.PV
+    name : str, optional
+        Name of signal.  If not given defaults to read_pv
+    put_complete : bool, optional
+        Use put completion when writing the value
+    '''
+    def __init__(self, read_pv, write_pv=None, *, pv_kw=None,
+                 put_complete=False, string=False, limits=False,
+                 auto_monitor=None, name=None, **kwargs):
+
+        if 'rw' in kwargs:
```

Comment:
this check could be moved up to the base-class init?  Given that this class is the default behavior, my guess would be that there are far more read-only signals with the `rw=False` kwarg.